src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.Infrastructure/Data/Repositories/PaymentRepository.cs

```diff
@@ -14,5 +14,25 @@ public void AddMany(List<PaymentEntity> payments, PaymentSchema schema)
         {
             ExecuteBatch(payments.ToArray(), $"{schema.ToString()}.Payments");
         }
+
+        public IEnumerable<PaymentEntity> GetAllHistoricPaymentsForProvider(long ukprn)
+        {
+            const string sql = @"
+            SELECT 
+                [LearnRefNumber]
+                ,[DeliveryMonth]
+                ,[DeliveryYear]
+                ,[Amount]
+                ,[RequiredPaymentId]
+                ,[CollectionPeriodName]
+                ,[CollectionPeriodMonth]
+                ,[CollectionPeriodYear]
+                ,[FundingSource]
+                ,[TransactionType]
+            FROM Reference.PaymentsHistory
+            WHERE Ukprn = @ukprn";
+
```

Comment:
Is it not worth using SQL to aggregate here? Do we really need to know all of this information? I would have thought that Ukprn, LearnRefNumber and SUM(Amount) for all funding source 3, transaction type 1/3 would be fine

Reply:
Trying not to put any logic into the SQL scripts, Also because it's in a shared repo, it may end up being used by another component

Reply:
I would call the method something different. Like EmployerPayments or something along those lines so that it's obvious what it's doing

Reply:
ok, will do